### DISCUSSION TITLE
My challenges

### DISCUSSION THREAD

**Discussion by sipper69** [sipper69/Control3#1] [Announcements]
I had a couple of challenges.

- lag, when the video is too slow your mouse pointer suffers lag, hard to position your mouse
- relative mouse movement, Linux does not support absolute mouse positioning (edit: it appears my RPi Linux does though!)
- overload of mouse movements, had to find a way to skip movements while serial processing
- disable local mouse pointer, not possible, just move it aside and keep it there
- fullscreen, how to exit, middle mouse button!
- NUM lock, still have to figure it out, but never use it on my laptop, is legacy now?
- CPU, through the roof, was hard to find GPU support, but WinRT API's rock! WinUI3 is soo fast
- compatibility, force MJPG to make it work on my different laptops
- packaging, WinUI3 and packaging is hell, I used third party. "unknown publisher" still there, code signing costs a fortune

Tackled them all, real low CPU and sometimes you don't even notice you're working remote :-)

**Comment by sipper69** [comment]
Put it on Microsoft App Store, no separate package signing needed.

**Reply by MexiKing76** [reply]
I have the new version and RDP does work much better now. I do get that lag on the RDP session but on the actual laptop I am controlling it responds in real-time. 

What was your NUM lock issue? It doesn't work consistently? 

I have another scenario for you. I also use RustDesk for remote access. When I use that the mouse works great, however their is no video I just see a white screen and when I click on the gear next to the Display the app closes.

**Reply by MexiKing76** [reply]
I think I see why RustDesk does not work, it supports these VP8 / VP9 / AV1 software codecs, and H264 / H265 hardware codecs.

**Reply by sipper69** [reply]
Nice, you used absolute pointer positioning, right?

Lag on your RDP session is just bandwidth related? Nothing to cache or handle efficiently in a high resolution, high color depth video :-)

Did you try lowering your resolution and color depth?
On my local wifi no issues with lag at all. 

Honestly, I don't really remember what my challenges with NUM-lock were. I assume it is a local action, ie. keycode of key "1" from the num pad is probably the same as the "1" on the left. I think the laptop I tested could not switch the num-lock during a session. 

Don't know RustDesk, you make me curious. 
Clicking settings causing the app to close is really weird, I'll check that. App closes when the USB cable is pulled, but that is not triggered by the settings option.

Interesting that a remote management tool is "processing" a video stream from the host. Did you try another encoding? NV12 or YUV2 (whatever your capture card supports). But if RustDesk does something smart with the videostream it could solve your lag issue!
